databus/src/main/java/com/bazaarvoice/emodb/databus/core/DefaultFanout.java

```diff
@@ -169,72 +183,91 @@ boolean copyEvents(List<EventData> rawEvents) {
+                        // 1. In-order delivery is not guaranteed
+                        // 2. The event time is based on the change ID which is close-to but not precisely the time the update occurred
+                        // 3. Injected events have artificial change IDs which don't correspond to any clock-based time
+                        // However, this is still a useful metric because:
+                        // 1. Delivery is in-order the majority of the time
+                        // 2. Change IDs are typically within milliseconds of update times
+                        // 3. Injected events are extremely rare and should be avoided outside of testing anyway
+                        // 4. The lag only becomes a concern on the scale of minutes, far above the uncertainty introduced by the above
+                        if (lastMatchEventData != null) {
+                            updateLagMetrics(lastMatchEventData.getEventTime());
                         }
                     }
+                } catch (Throwable t) {
+                    _log.error("Uncaught exception in fanout pool. Thread will die, but it should be replaced.", t);
+                    throw t;
                 }
-
-                // Flush to cap the amount of memory used to buffer events.
-                if (eventsByChannel.size() >= FLUSH_EVENTS_THRESHOLD) {
-                    flush(eventKeys, eventsByChannel, numOutboundReplicationEvents);
-                    numOutboundReplicationEvents = 0;
-                }
-
-                // Track the final match event data record returned
-                lastMatchEventData = matchEventData;
-            }
-
-            // Final flush.
-            flush(eventKeys, eventsByChannel, numOutboundReplicationEvents);
+            }));
         }
 
-        // Update the lag metrics based on the last event returned.  This isn't perfect for several reasons:
-        // 1. In-order delivery is not guaranteed
-        // 2. The event time is based on the change ID which is close-to but not precisely the time the update occurred
-        // 3. Injected events have artificial change IDs which don't correspond to any clock-based time
-        // However, this is still a useful metric because:
-        // 1. Delivery is in-order the majority of the time
-        // 2. Change IDs are typically within milliseconds of update times
-        // 3. Injected events are extremely rare and should be avoided outside of testing anyway
-        // 4. The lag only becomes a concern on the scale of minutes, far above the uncertainty introduced by the above
-        if (lastMatchEventData != null) {
-            updateLagMetrics(lastMatchEventData.getEventTime());
+        for (final Future<?> future : futures) {
+            try {
+                future.get();
+            } catch (InterruptedException | ExecutionException e) {
```

Comment:
ok

Reply:
Personally, I don't like that guava stuff. I'd much rather have 3 lines of boilerplate than an unnecessary dependency. Letting guava seduce you with all these trivial convenience libraries will make it borderline impossible to update or replace that library later on.

I think Cache and company are really awesome and helpful. But crap like `newArrayList()` instead of `new ArrayList()` is just silly and ultimately harmful.

For example, in this situation, `getUnchecked` actually does some bizarro wrapping and unwrapping of the exception, which I think will make it harder to handle the exception if you need to, whereas the "boilerplate" that I have written is very clear and obvious about what it does.

But it's not my project, so I'm following your recommendation.